### ISSUE TITLE
False positive `unnecessary-ellipsis` when multiple conditions in `if` statement

### ISSUE THREAD

**Issue by skshetry** [PyCQA/pylint#6071]
### Bug description

```python
# test.py
def func(val1, val2):
   if val1 is not ... and val2:
        pass
```

Above code raises error, but the following does not:
```python
def func(val1, _):
   if val1 is not ...:
        pass
```

So, it probably is something to do with multiple conditions here.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:2:18: W2301: Unnecessary ellipsis constant (unnecessary-ellipsis)
```

### Expected behavior

No false-positives

### Pylint version

```shell
pylint 2.13.4
astroid 2.11.2
Python 3.10.0 (default, Oct 17 2021, 20:19:40) [Clang 12.0.5 (clang-1205.0.22.9)]
```
